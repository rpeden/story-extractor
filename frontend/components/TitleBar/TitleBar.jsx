import React from "react";
import AppBar from "material-ui/AppBar";
import FlatButton from "material-ui/FlatButton";
import { observer } from "mobx-react";

export default class TitleBar extends React.Component {

    lastUpdated() {
        if (this.props.lastUpdated) {
        const buttonLabel = this.props.currentlyUpdating ? "Updating..." : "Update Now";

        return (
            <div className="title-bar-update-container">
                <div className="title-bar-update-text">
                    <span>
                    Last Updated: {this.props.lastUpdated}
                    </span>
                </div>
                <FlatButton
                    className="title-bar-update-button"
                    label={buttonLabel}
                    onClick={this.props.onUpdateClick}
                />
            </div>
        );
        }

        return <span />;
    }

    render() {
        return (
            <AppBar
                iconElementLeft={<span />}
                iconElementRight={this.lastUpdated()}
                title="Story Grabber 9000"
                className="title-bar"
            />
        );
    }
}