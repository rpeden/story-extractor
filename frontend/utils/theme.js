import getMuiTheme from "material-ui/styles/getMuiTheme";
import { red500, red700 } from "material-ui/styles/colors";
import injectTapEventPlugin from "react-tap-event-plugin";

injectTapEventPlugin();

const muiTheme = getMuiTheme({
  palette: {
    primary1Color: red500,
    primary2Color: red700
  }
});

export default muiTheme;
