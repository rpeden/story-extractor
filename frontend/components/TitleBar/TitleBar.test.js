import React from "react";
import renderer from "react-test-renderer";
import {mount} from "enzyme";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import theme from "../../utils/theme";

import TitleBar from "./TitleBar.jsx";

test("TitleBar snapshot", () => {
  const component = renderer.create(
    <MuiThemeProvider theme={theme}>
      <TitleBar lastUpdated={"now"} />
    </MuiThemeProvider>
  );
  const titleBar = component.toJSON();
  expect(titleBar).toMatchSnapshot();
});

test("Last updated time displays correctly", () => {
  const component = mount(
    <MuiThemeProvider theme={theme}>
      <TitleBar lastUpdated={"now"} />
    </MuiThemeProvider>
  );

  const updatedText = component.find(".title-bar-update-text").text();
  expect(updatedText).toBe("Last Updated: now");
});
