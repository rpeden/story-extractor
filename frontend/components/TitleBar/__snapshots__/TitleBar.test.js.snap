// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar snapshot 1`] = `
<div
  className="title-bar"
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#00bcd4",
      "borderRadius": "0px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "paddingLeft": 24,
      "paddingRight": 24,
      "position": "relative",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1100,
    }
  }
>
  <div
    style={
      Object {
        "marginLeft": -16,
        "marginRight": 8,
        "marginTop": 8,
        "muiPrepared": true,
      }
    }
  >
    <span />
  </div>
  <div
    onTouchTap={[Function]}
    style={
      Object {
        "boxFlex": 1,
        "color": "#ffffff",
        "flex": "1",
        "fontSize": 24,
        "fontWeight": 400,
        "height": 64,
        "letterSpacing": 0,
        "lineHeight": "64px",
        "margin": 0,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingTop": 0,
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
  >
    <div
      style={
        Object {
          "fontFamily": "Orbitron",
        }
      }
    >
      StoryGrabber
    </div>
  </div>
  <div
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": -16,
        "marginTop": 8,
        "muiPrepared": true,
      }
    }
  >
    <div
      className="title-bar-update-container"
    >
      <div
        className="title-bar-update-text"
      >
        <span>
          Last Updated: 
          now
        </span>
      </div>
    </div>
  </div>
</div>
`;
